pkg/operator/ceph/csi/network.go

```diff
@@ -0,0 +1,94 @@
+package csi
+
+import (
+	_ "embed"
+	"fmt"
+
+	"github.com/pkg/errors"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	opcontroller "github.com/rook/rook/pkg/operator/ceph/controller"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	corev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	multusDaemonsetName = "csi-multus"
+	multusLabel         = "rook-ceph-multus"
+	multusFinalizer     = "multus.ceph.rook.io"
+)
+
+var (
+	//go:embed template/multus/daemonset.yaml
+	MultusDaemonsetTemplatePath string
+)
+
+func (r *ReconcileCSI) setupCSINetwork(cephCluster cephv1.CephCluster) error {
+	if !cephCluster.Spec.Network.IsMultus() {
+		logger.Debug("multus not used; no additional network configuration necessary.")
+		return nil
+	}
+
+	if err := opcontroller.AddFinalizerWithNameIfNotPresent(r.opManagerContext, r.client, &cephCluster, multusFinalizer); err != nil {
```

Comment:
Even if there are multiple resources to cleanup when a cluster CR is deleted, we only need a single finalizer so the operator can trigger all the cleanup code, so I don't believe we need this new finalizer.